### ISSUE TITLE
random-id footgun in Bucket creation

### ISSUE THREAD

**Issue by rienafairefr** [pulumi/pulumi-gcp#698]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details
I was trying to make a bucket with a domain name name. Let's say auth.example.com, tried to do it like this:

```
bucket = storage.Bucket("auth.example.com")
```

And I had a `* googleapi: Error 400: Bucket name contains ".", but is not under a currently recognized top-level domain., invalid` during the creation.
I tried to troubleshoot and find the culprit, the domain was validated owned by me, pulumi was using my user and not a service account, high and low I searched.

Couldn't really find a way to easily see the HTTP calls made by the "google-api-go-client/0.5 Pulumi/3.0 (GPN:Pulumi; https://www.pulumi.com) pulumi-gcp/v5.23.0,gzip(gfe)" user agent run by pulumi. no amount of looking at https://www.pulumi.com/docs/troubleshooting/ did it

I managed to find the HTTP calls by looking at the audit logs for storage.googleapis.com and looking at the call that pulumi was making with the storage.buckets.create permission. I found this in the payload

`resource: "projects/_/buckets/auth.example.com-3da1b83"`

pulumi, helpfully, was adding a random id at the end, and auth.example.com-3da1b83 is not a domain name owned by me, of course. It's not even a valid domain name (google should probably tell us that instead of focusing on whether or not we own it, but that's not pulumi's fault)

Of course, problem between chair and computer. Changing the config to force the bucket name to be exactly the domain name, it works:
```
bucket = storage.Bucket("auth_example_com", name="auth.example.com")
```

TLDR:
Bucket Creation adds a random id at the end. Which is fine in general, but deadly (and difficult to see-deadly) when creating a bucket with a domain name.

**Comment by leezen** [comment]
Pulumi's autonaming behavior is documented in https://www.pulumi.com/docs/intro/concepts/resources/#names -- I'd love to hear yours thoughts on ways this could have been made clearer as part of your experience using Pulumi as we'd like to obviously help people avoid hitting unexpected behavior.
